pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -488,10 +491,23 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 		}
 	} else if spec.RancherKubernetesEngineConfig != nil && spec.RancherKubernetesEngineConfig.Restore.Restore {
 		logrus.Infof("Restoring cluster [%s] from backup", cluster.Name)
+		// ensure restore does not get short-circuited by key rotation, restores may be performed if a key rotation fails
```

Comment:
since it's connected to RKE's code, it'd help to mention that here: either the code line or that RKE checks for rotateEncryptionKey first (for later changes)

Reply:
Yes, I'll expand on that in the comment!